tests/common/nebula_test_suite.py

```diff
@@ -351,18 +352,97 @@ def to_value(self, col):
         elif isinstance(col, str):
             value.set_sVal(col.encode('utf-8'))
         elif isinstance(col, dict):
-            map_val = CommonTtypes.Map()
-            map_val.kvs = dict()
-            for key in col:
-                ok, temp = self.to_value(col[key])
+            if "type" not in col:
+                map_val = CommonTtypes.Map()
+                map_val.kvs = dict()
+                for key in col:
+                    ok, temp = self.to_value(col[key])
+                    if not ok:
+                        return ok, temp
+                    map_val.kvs[key.encode('utf-8')] = temp
+                value.set_mVal(map_val)
+            else:
+                ok, temp = self.convert_vertex_edge(col)
                 if not ok:
-                    return ok, temp
-                map_val.kvs[key.encode('utf-8')] = temp
-            value.set_mVal(map_val)
+                    return False, 'Wrong val type'
+                value.set_lVal(temp)
         else:
             return False, 'Wrong val type'
         return True, value
 
+    @classmethod
+    def convert_to_vertex(self, v):
+        vertex = CommonTtypes.Vertex()
+        vertex.vid = bytes(v["vid"], encoding = 'utf-8')
+        tags = []
+        for tagVal in v["tags"]:
+            tag = CommonTtypes.Tag()
+            tag.name= bytes(tagVal["name"], encoding = 'utf-8')
+            props = dict()
+            for k, v in tagVal["props"].items():
+                value = CommonTtypes.Value()
+                if isinstance(v, int):
```

Comment:
The type is not only int/float/string. you can use `self.to_value` directly.

Reply:
done